src/odin/ingestion/afc/afc_archive.py

```diff
@@ -214,23 +224,36 @@ def setup_job(self) -> None:
         # set self.pq_job_id from parquet dataset
         self.pq_job_id: int | None = None
         self.max_job_id: int | None = None
-        pq_objects = list_objects(self.export_folder, in_filter=".parquet")
-        if pq_objects:
+        if list_objects(self.export_folder, in_filter=".parquet"):
             _, self.pq_job_id = ds_metadata_min_max(
                 ds_from_path(f"s3://{self.export_folder}"), "job_id"
             )
-        log.add_metadata(pq_job_id=self.pq_job_id)
+        log.complete(pq_job_id=self.pq_job_id)
 
-        # pull list of job_id's available to process from API
-        count_url = f"{API_ROOT}/count"
-        count_fields = {"table_name": self.table}
-        if self.pq_job_id is not None:
-            count_fields["jobIdFrom"] = str(self.pq_job_id)
-        r = self.make_request(count_url, fields=count_fields)
-        self.job_ids: APICounts = r.json()
-        assert isinstance(self.job_ids, list)
+    def api_job_ids(self) -> Generator[APICounts]:
+        """
+        Generate job_id's from /count endpoint.
 
-        log.complete(num_job_ids=len(self.job_ids))
+        Will pull job_ids from /count endpoint in batches until no more job_ids are available.
+        Initial request will be exclusive of self.pq_job_id
+        """
+        count_url = f"{API_ROOT}/count"
+        req_fields = {"table_name": self.table, "limit": str(10_000)}
+        job_id_from = self.pq_job_id
+        while True:
+            if job_id_from is not None:
+                req_fields["jobIdFrom"] = str(job_id_from + 1)
+            try:
+                r = self.make_request(count_url, fields=req_fields)
+            except APIEmptyResponse:
+                break
+            api_counts: APICounts = r.json()
+            api_counts = sorted(api_counts, key=itemgetter("jobId"))
+            if api_counts[-1]["jobId"] != job_id_from:
+                yield api_counts
+            else:
```

Comment:
This `else` probably will never be reached with the new logic to add 1 for the jobIdFrom parameter. We may only ever break on the `APIEmptyResponse` exception.